internal/controller/clusterinstance/helper.go

```diff
@@ -124,7 +124,10 @@ func getInstallConfigOverrides(clusterInstance *v1alpha1.ClusterInstance) (strin
 }
 
 // buildClusterData returns a Cluster object that is consumed for rendering templates
-func buildClusterData(clusterInstance *v1alpha1.ClusterInstance, node *v1alpha1.NodeSpec,
+func buildClusterData(
+	clusterInstance *v1alpha1.ClusterInstance,
+	node *v1alpha1.NodeSpec,
+	clusterImageSet *hivev1.ClusterImageSet,
```

Comment:
Why do you need to pass the `ClusterImageSet` as an argument? You can rather pass the kube client and execute `c.Get(ctx, key, &clusterImageSet)` in this function.

Reply:
That's a valid point, however, we can address the efficiency aspect by using a cached client in a separate PR.